ablySpec/Auth.swift

```diff
@@ -599,5 +599,406 @@ class Auth : QuickSpec {
+                authOptions.force = true
+
+                // Force new token
+                waitUntil(timeout: testTimeout) { done in
+                    rest.auth.authorise(nil, options: authOptions) { tokenDetails, error in
+                        expect(error).to(beNil())
+                        expect(tokenDetails?.token).toNot(equal(currentToken))
+                        done()
+                    }
+                }
+            }
+
+            // RSA10e
+            it("should use the requestToken implementation") {
+                let rest = ARTRest(options: AblyTests.commonAppSetup())
+
+                var requestMethodWasCalled = false
+                let block: @convention(block) (AspectInfo) -> Void = { _ in
+                    requestMethodWasCalled = true
+                }
+
+                let hook = ARTAuth.aspect_hookSelector(rest.auth)
+                // Adds a block of code after `requestToken` is triggered
+                let token = try? hook("requestToken:withOptions:callback:", withOptions: .PositionAfter, usingBlock:  unsafeBitCast(block, ARTAuth.self))
+
+                expect(token).toNot(beNil())
+
+                waitUntil(timeout: testTimeout) { done in
+                    rest.auth.authorise(nil, options: nil, callback: { tokenDetails, error in
+                        expect(error).to(beNil())
+                        expect(tokenDetails?.token).toNot(beEmpty())
+                        done()
+                    })
+                }
+
+                expect(requestMethodWasCalled).to(beTrue())
+            }
+
+            func expectTokenDetails(tokenDetails: ARTAuthTokenDetails?) {
+                expect(tokenDetails).toNot(beNil())
+                expect(tokenDetails).to(beAnInstanceOf(ARTAuthTokenDetails))
+                expect(tokenDetails?.token).toNot(beEmpty())
+                expect(tokenDetails?.issued).toNot(beNil())
+                expect(tokenDetails?.expires).toNot(beNil())
+                expect(tokenDetails?.expires?.timeIntervalSince1970).to(beGreaterThan(tokenDetails?.issued?.timeIntervalSince1970))
+            }
+
+            // RSA10f
+            it("should return TokenDetails with valid token metadata") {
+                let options = AblyTests.commonAppSetup()
+                options.clientId = "client_string"
+                let rest = ARTRest(options: options)
+
+                waitUntil(timeout: testTimeout) { done in
+                    rest.auth.authorise(nil, options: nil) { tokenDetails, error in
+                        expect(error).to(beNil())
+                        expectTokenDetails(tokenDetails)
+                        expect(tokenDetails?.clientId).to(equal(options.clientId))
+                        done()
+                    }
+                }
+            }
+
+            // RSA10g
+            it("should store the AuthOptions and TokenParams arguments as defaults for subsequent authorisations") {
```

Comment:
This test is good, but could we add additional attributes please such as `ttl` and `capability` as all other tests seem to only be testing one attribute `clientId`